raster-foundry/raster-foundry
# Adjust styles to handle custom checkbox icons

## Overview

This PR makes some adjustments to the `toggle` component and the associated styles.

By default, the new toggle component displays a checkbox when no content is wrapped:

```html
<rf-toggle value="$ctrl.isCurrentProcessingOption(option)"
           on-change="$ctrl.handleOptionChange(value, option)">
</rf-toggle>
```

You can also opt-in to a custom icon by setting the `custom-icon` parameter to `true` and putting the icon you want in the toggle:

```html
<rf-toggle value="$ctrl.isCurrentProcessingOption(option)"
           on-change="$ctrl.handleOptionChange(value, option)"
           custom-icon="true">
    <i class="icon-cross"></i>
</rf-toggle>
```

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

No icon specified:
![image](https://user-images.githubusercontent.com/2442245/34311616-8f4b4880-e72c-11e7-992a-c6d1f7273497.png)


Custom icon being used:
![image](https://user-images.githubusercontent.com/2442245/34311597-77b0f15c-e72c-11e7-9a43-c732f088921b.png)


## Testing Instructions

 * Test some of the uses of the `toggle` component to see that they work as intended.

Closes #2761
